src/map.rs

```diff
@@ -2775,14 +2753,14 @@ where
                     let mut head = Shared::null();
                     let mut tail = Shared::null();
                     while !e.is_null() {
-                        // safety: either `e` is `bin`, in which case it is valid due to the above,
-                        // or `e` was obtained from a next pointer. Any next pointer obtained from
-                        // bin is valid at the time we look up bin in the table, at which point the
-                        // epoch is pinned by our guard. Since we found the next pointer in a valid
-                        // map and it is not null (as checked above), the node it points to was
-                        // present (i.e. not removed) from the map in the current epoch. Thus,
-                        // because the epoch cannot advance until we release our guard, `e` is also
-                        // valid if it was obtained from a next pointer.
+                        // safety: either e is bin, in which case it is valid due to the above,
+                        // or e was obtained from a next pointer. Any next pointer obtained from
+                        // bin is valid at the time we look up bin in the table, at which point
+                        // we held a guard. Since we found the next pointer in a valid map and
+                        // it is not null (as checked above and below), the node it points to was
+                        // present (i.e. not removed) from the map while we were marked as active.
+                        // Thus, e cannot be dropped until we release our guard, e is also valid
```

Comment:
```suggestion
                        // Thus, e cannot be dropped until we release our guard, and so e is also valid
```

Reply:
The rest of the comment says "e is also valid if it was obtained from a next pointer." Maybe change it to a period instead of a comma?